Firestore/core/src/local/memory_remote_document_cache.cc

```diff
@@ -87,7 +87,8 @@ MutableDocumentMap MemoryRemoteDocumentCache::GetAll(
       break;
     }
     const MutableDocument& document = it->second;
-    if (key.path().size() > path.size() + 1) {
+    size_t immediate_children_path_length = path.size() + 1;
```

Comment:
Please define `immediate_children_path_length` outside of the `for` loop to avoid recalculating it all the time. This also improves readability because its "const-ness" is correctly implied by being outside of the `for` loop.